### ISSUE TITLE
Add indentation metadata to om.dom and other om vars

### ISSUE THREAD

**Issue by levitanong** [omcljs/om#728]
https://github.com/clojure-emacs/cider/issues/747

With the addition of
>```clj
(defn ^:private gen-react-dom-fn [tag]
  `(defn ~tag [opts# & children#]
     {:style/indent 1}
     (.apply ~(symbol "js" (str "React.DOM." (name tag))) nil
       (cljs.core/into-array
         (cons opts# (cljs.core/map om.util/force-children children#))))))
```

All om dom elements can indent properly and automatically in emacs. (and possibly future IDEs.)

**Comment by anmonteiro** [comment]
FWIW, I don't think editor specific information should be added like this.

You can probably achieve the same functionality in emacs with the following:

```emacs-lisp
(setq clojure-indent-style :always-indent)
```

**Comment by levitanong** [comment]
I did have some doubts, but as it happens, cider and clojure-mode communities have talked about how to approach indentation.

At the moment, I keep a custom list of vars and how they should be indented. I'm fine with just maintaining that list and growing it as I use things, but I figured it was worth a shot.

The functionality is quite different, as the desired result is a macro-like indentation for om components (basically constant 2-space indent)...
```clj
(dom/div opts
  (dom/div nil "child")
  (dom/div nil "derp))
```

...while maintaining the "idiomatic" clojure indentation for all other forms.
```clj
(some-fn arg1
         arg2)
```
